lms/djangoapps/courseware/date_summary.py

```diff
@@ -444,8 +443,6 @@ class CourseExpiredDate(DateSummary):
 
     @property
     def date(self):
-        if not CourseDurationLimitConfig.enabled_for_enrollment(self.user, self.course):
-            return
```

Comment:
This and related changes were just to pull this bit of extra context (that was used everywhere `get_user_course_expiration_date` and `get_user_course_duration` were called) into those methods. Felt dumb that every caller had to know about this flag.